File: source/manual/latex-original/ds_writing/writing/steppermotorclass1.cpp
#include <tango.h>

#include <steppermotor.h>
#include <steppermotorclass.h>

namespace StepperMotor
{

StepperMotorClass *StepperMotorClass::_instance = NULL;

StepperMotorClass::StepperMotorClass(string &s):
Tango::DeviceClass(s)
{
	cout2 << "Entering StepperMotorClass constructor" << endl;
	
	cout2 << "Leaving StepperMotorClass constructor" << endl;
}


StepperMotorClass *StepperMotorClass::init(const char *name)
{
	if (_instance == NULL)
	{
		try
		{
			string s(name);
			_instance = new StepperMotorClass(s);
		}
		catch (bad_alloc)
		{
			throw;
		}		
	}		
	return _instance;
}

StepperMotorClass *StepperMotorClass::instance()
{
	if (_instance == NULL)
	{
		cerr << "Class is not initialised !!" << endl;
		exit(-1);
	}
	return _instance;
}

File: source/development/device-api/device-server-writing/writing_java/StepperMotorClass1.java
package StepperMotor;

import java.util.*;
import fr.esrf.Tango.*;
import fr.esrf.TangoDs.*;

public class StepperMotorClass extends DeviceClass implements TangoConst
{
	private static StepperMotorClass 	_instance = null;
	
	
	public static StepperMotorClass instance()
	{
		if (_instance == null)
		{
			System.err.println("StepperMotorClass is not initialised !!!");
			System.err.println("Exiting");
			System.exit(-1);
		}
		return _instance;
	}


	public static StepperMotorClass init(String class_name) throws DevFailed
	{
		if (_instance == null)
		{
			_instance = new StepperMotorClass(class_name);
		}
		return _instance;
	}
	
	protected StepperMotorClass(String name) throws DevFailed
	{
		super(name);

		Util.out2.println("Entering StepperMotorClass constructor");
	
		Util.out2.println("Leaving StepperMotorClass constructor");
	}
}
